src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -1082,7 +1087,42 @@ namespace QgsWms
       return styleElem;
     }
 
-    void appendLayersFromTreeGroup( QDomDocument &doc, QDomElement &parentLayer, QgsServerInterface *serverIface, const QgsProject *project, const QgsWmsRequest &request, const QgsLayerTreeGroup *layerTreeGroup, const QMap<QString, QgsWmsLayerInfos> &wmsLayerInfos, bool projectSettings )
+    //! Return TRUE if date only have been written, FALSE if there is datetime
+    bool writeTimeDimensionNode( QDomDocument &doc, QDomElement &layerElem, const QList<QgsDateTimeRange> &dateRanges )
+    {
+      // Apparently, for vectors allTemporalRanges is always empty :/
+      // there is no way to know the type of range or the individual instants
+
+      // we write a TIME dimension even if dateRanges is empty. Not sure this is appropriate but
+      // it was like that from the beginning so better keep it that way to avoid regression on client side
+
+      const bool hasDateTime = std::any_of( dateRanges.constBegin(), dateRanges.constEnd(), []( const QgsDateTimeRange &r ) { return r.begin().time() != QTime( 0, 0 )
+                                                                                                                                     || ( !r.isInstant() && r.end().time() != QTime( 0, 0 ) ); } );
+
+      const QString dateFormat = hasDateTime ? QStringLiteral( "yyyy-MM-ddTHH:mm:ss" ) : QStringLiteral( "yyyy-MM-dd" );
+
+      QStringList strValues;
+      for ( const QgsDateTimeRange &r : dateRanges )
```

Comment:
could you please rename the variable to avoid single length?

Reply:
:+1: